staff_features/accessions/step_definitions/accessions_search.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+Given 'an Accession has been created' do
+  @uuid = SecureRandom.uuid
+
+  visit "#{STAFF_URL}/accessions/new"
+  fill_in 'accession_id_0_', with: "Accession #{@uuid}"
+  click_on 'Save'
+end
+
+Then 'the Accession is in the search results' do
+  expect(page).to have_css('tr', text: @uuid)
+end
+
+Then 'the Accession view page is displayed' do
+  expect(find('h2').text).to eq "Accession #{@uuid} Accession"
+end
+
+Given 'two Accessions have been created with a common keyword in their title' do
+  @shared_accession_uuid = SecureRandom.uuid
+  @accession_a_uuid = SecureRandom.uuid
+  @accession_b_uuid = SecureRandom.uuid
+
+  visit "#{STAFF_URL}/accessions/new"
+  fill_in 'accession_title_', with: "Accession A #{@accession_a_uuid} #{@shared_accession_uuid}"
+  fill_in 'accession_id_0_', with: "Accession A #{@accession_a_uuid}"
+  click_on 'Save'
+
+  visit "#{STAFF_URL}/accessions/new"
+  fill_in 'accession_title_', with: "Accession B #{@accession_b_uuid} #{@shared_accession_uuid}"
+  fill_in 'accession_id_0_', with: "Accession B #{@accession_b_uuid}"
+  click_on 'Save'
+end
+
+When 'the user filters by text with the common title keyword used by both Accessions' do
+  visit "#{STAFF_URL}/accessions"
+
+  fill_in 'filter-text', with: @shared_accession_uuid
+
+  within '.search-filter' do
+    find('button').click
+  end
+
+  search_result_rows = all('#tabledSearchResults tbody tr')
+
+  expect(search_result_rows.length).to eq 2
+  expect(search_result_rows[0]).to have_text @accession_a_uuid
+  expect(search_result_rows[1]).to have_text @accession_b_uuid
```

Comment:
Should the When function have the Then tests?

Reply:
Good point. These expectations are part of the state, rather than a result of the actions of the user. In the pure BDD approach, as usually implemented with `RSpec`, expectations are always a result of previous actions, taking place in a pre-determined state.

However, in the context of `Cucumber` with `Capybara` with end-to-end testing, we may use `expect` assertions anywhere inside the step definitions (Given, When, Then), to ensure the integrity and stability of the tests, in order to avoid non-deterministic test failures (flaky tests). So an `expect` from `RSpec` does not necessarily match a `Then` expectation from the `Gherkin` features.

Here we could consider these statements as part of the pre-requisites for this Scenario. These expectations probably belong to a `Given` block. I have already discussed this with @kdimopulu, and refactoring is going to take place.

We will try to avoid `expect` as much as possible from `Then` step definitions, however, we may encounter occasions where `expect`s have to be present in a `Then` block to make `Capybara` wait for elements or pages/views to appear, finish AJAX requests, etc.

Reply:
I think this makes sense although I think maybe my question was also not quite clear. I wasn't really questioning the use of `expect` assertions generally, but rather the fact that in this case, these 3 expect statements appear in both the When and the Then clause identically which seemed a little strange to me.

Reply:
Oh, ok. It is quite clear now. These expectations are slightly different.
The first two ensure that the results are sorted by ascending title before the user clicks on `Title`, while the other two are used to ensure descending order after the user action.

The difference lies in `([0], [1]) -> ([1], [0])`

Results in ascending order:
```
expect(search_result_rows.length).to eq 2
expect(search_result_rows[0]).to have_text @accession_a_uuid
expect(search_result_rows[1]).to have_text @accession_b_uuid
```

Results in descending order:
```
expect(search_result_rows.length).to eq 2
expect(search_result_rows[1]).to have_text @accession_a_uuid
expect(search_result_rows[0]).to have_text @accession_b_uuid
```